### PR TITLE
reality: Get version info from reality store.

### PR DESCRIPTION
**Pull request by petertseng**

### CODE REVIEW

**Comment by tummychow** [comment]
lgtm in general, but it's been a while since I looked at the semantics of this package. You might want to take a look at this as well @anthonybishopric

File: pkg/health/health.go [hunk 1]
```diff
@@ -66,18 +67,31 @@ func (s *ServiceNodeStatus) IsCurrentVersion(version string) bool {
...
 		Health: consulHealth,
+		Store:  store,
 	}
 }
 
-func (s *ConsulHealthChecker) toNodeStatus(entry consulapi.ServiceEntry) *ServiceNodeStatus {
+func (s *ConsulHealthChecker) toNodeStatus(serviceID string, entry consulapi.ServiceEntry) *ServiceNodeStatus {
+
+	version := "unknown"
+	manifest, _, _ := s.Store.Pod(kp.RealityPath(entry.Node.Node, serviceID))
```
Comment (anthonybishopric) [inline pkg/health/health.go]: Can't there be an error here?

File: pkg/health/health.go [hunk 2]
```diff
@@ -66,18 +67,31 @@ func (s *ServiceNodeStatus) IsCurrentVersion(version string) bool {
...
+	if manifest != nil {
+		var err error
+		version, err = manifest.SHA()
+		if err != nil {
+			version = "unknown"
```
Comment (anthonybishopric) [inline pkg/health/health.go]: make a const

File: pkg/health/health.go (hunk 1 above)
Comment (petertseng) [inline pkg/health/health.go]: the manifest would be nil in that case, wouldn't it? But I'll check the err as well.

File: pkg/health/health.go (hunk 2 above)
Comment (petertseng) [inline pkg/health/health.go]: hmm is 4ead91a what you are looking for

File: pkg/health/health.go [hunk 3]
```diff
@@ -66,18 +69,30 @@ func (s *ServiceNodeStatus) IsCurrentVersion(version string) bool {
...
+		Store:  store,
 	}
 }
 
-func (s *ConsulHealthChecker) toNodeStatus(entry consulapi.ServiceEntry) *ServiceNodeStatus {
+func (s *ConsulHealthChecker) toNodeStatus(serviceID string, entry consulapi.ServiceEntry) *ServiceNodeStatus {
+
+	version := UnknownSHA
+	manifest, _, err := s.Store.Pod(kp.RealityPath(entry.Node.Node, serviceID))
+	if manifest != nil && err == nil {
```
Comment (anthonybishopric) [inline pkg/health/health.go]: well, more specifically, we should probably know if this fails, either by returning the error or something smarter.

**Comment by petertseng** [comment]
Now returning an error from toNodeStatus

File: pkg/health/health.go [hunk 4]
```diff
@@ -66,25 +67,41 @@ func (s *ServiceNodeStatus) IsCurrentVersion(version string) bool {
...
+		Store:  store,
 	}
 }
 
-func (s *ConsulHealthChecker) toNodeStatus(entry consulapi.ServiceEntry) *ServiceNodeStatus {
+func (s *ConsulHealthChecker) toNodeStatus(serviceID string, entry consulapi.ServiceEntry) (*ServiceNodeStatus, error) {
+
+	version := ""
+	manifest, _, err := s.Store.Pod(kp.RealityPath(entry.Node.Node, serviceID))
+	if err != nil {
```
Comment (petertseng) [inline pkg/health/health.go]: Wondering whether I should handle `pods.NoCurrentManifest` here specially, such as `err != nil && err != pods.NoCurrentManifest`

Comment (petertseng) [inline pkg/health/health.go]: yeah doing it

File: pkg/health/health.go [hunk 5]
```diff
@@ -96,12 +114,18 @@ func (s *ConsulHealthChecker) LookupHealth(serviceID string) (*ServiceStatus, er
...
 		Statuses: make(map[string]*ServiceNodeStatus),
 	}
+
+	var firstErr error = nil
 	for _, entry := range entries {
 		if serviceID == entry.Service.ID {
-			serviceStatus.Statuses[entry.Node.Node] = s.toNodeStatus(*entry)
+			status, err := s.toNodeStatus(serviceID, *entry)
+			if err != nil && firstErr == nil {
+				firstErr = err
```
Comment (anthonybishopric) [inline pkg/health/health.go]: I'm still feeling the tingle of error suppression but this will at least signal that things are messed up.

**Comment by anthonybishopric** [comment]
:+1:

**Pull request merged**